frontend-react/src/components/Table/Table.test.tsx

```diff
@@ -1,28 +1,48 @@
-import { fireEvent, screen } from "@testing-library/react";
+import { fireEvent, render, screen } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
+import { act } from "react-dom/test-utils";
 
 import { renderWithRouter } from "../../utils/CustomRenderUtils";
 import { mockFilterManager } from "../../hooks/filters/mocks/MockFilterManager";
 import { SortSettingsActionType } from "../../hooks/filters/UseSortOrder";
 
 import { TestTable } from "./TestTable";
-import Table, { ColumnConfig, TableConfig, TableRow } from "./Table";
+import Table, {
+    ColumnConfig,
+    TableConfig,
+    TableRow,
+    TableRows,
+    ColumnData,
+} from "./Table";
 
 /* Table generation tools */
 
-const getSetOfRows = (count: number) => {
+const getSetOfRows = (count: number, linkable: boolean = true) => {
     const testRows: TableRow[] = [];
     for (let i = 0; i < count; i++) {
-        testRows.push({
-            id: i,
-            item: `Item ${i}`,
-            linkableItem: `UUID-${i}`,
-            mappedItem: i,
-            transformedValue: i,
-            sortedColumn: i,
-            locallySortedColumn: i,
-            editableColumn: i,
-        });
+        // this is bad, but I couldn't figure out how to do it another way
+        // without jumping through lots of typescript hoops - DWS
+        const row = linkable
```

Comment:
so, here what I'm trying to do is allow for a row to be created without a linked column. I tried both first creating an object without the `linkableItem` attribute, then adding it if the linkable flag came through, as well as creating an object with the attribute and deleting it if the flag didn't come through. In both cases, typescript yelled at me. I think because the TableRow is loosely defined, once you define an object of that type, typescript likes the general signature to remain as is. The workaround I found was this, just creating two different objects depending on the situation rather than mutating one. There's probably a better way, but I couldn't find it.